python/ray/tune/experiment/trial.py

```diff
@@ -636,7 +636,13 @@ def local_experiment_path(self, local_path: str):
             # relative to the old `local_dir`/`logdir`
             for checkpoint in self.get_trial_checkpoints():
                 checkpoint_dir = checkpoint.dir_or_data
-                assert isinstance(checkpoint_dir, str)
+                if not isinstance(checkpoint_dir, str):
```

Comment:
nit: Should we also support `Path` here?

Reply:
These are all loaded and internally generated checkpoints, and there are only 3 valid types: None, ray.ObjectRef and string. So we don't have to support this here.